### DISCUSSION TITLE
How to get list of roles

### DISCUSSION THREAD

**Discussion by deathemperor** [hasura/graphql-engine#6888] [Question]
I try looking around but couldn't find a way to get the list of roles. The "admin, user" roles that is.

![image](https://user-images.githubusercontent.com/4255482/117636230-cd308d80-b1aa-11eb-84ad-3fc6405c84c9.png)

Thanks!

**Comment by wwwdepot** [comment]
If you check the network logs, following is how the console gets the list of roles.

```bash
curl --request POST \
  --url http://YOUR_HASURA/v1/query \
  --header 'Content-Type: application/json' \
  --header 'x-hasura-admin-secret: YOUR_HASURA_ADMIN_KEY' \
  --data '{"type":"select","args":{"table":{"schema":"hdb_catalog","name":"hdb_role"},"columns":["role_name"],"order_by":[{"column":"role_name","type":"asc"}]}}'
```

**Reply by deathemperor** [reply]
thanks. it's useful but I really dont wanna query directly into the database.
